guides/hack/04-expressions-and-operators/102-banning-lval-as-an-expression.md

```diff
@@ -0,0 +1,64 @@
+## Why?
+
+1. We'd like to have various rules preventing things we know is unsafe in concurrent blocks. For example, assigning something twice in the middle of two different expressions in a concurrent block. `concurrent { $x1 = await genx($y = 42); $x2 = await geny($y = 43); }`
```

Comment:
*are unsafe

Reply:
also, throughout avoid the genx prefix; this is currently considered a flibism, and doesn't really make sense without the historical context of faking awaitables with generators